docs/user-guide/docker-cli-to-kubectl.md

```diff
@@ -0,0 +1,252 @@
+NAME         LABELS          SELECTOR        IP(S)     PORT(S)
+nginx-http   run=nginx-app   run=nginx-app             80/TCP
+```
+
+With kubectl, we create a [replication controller](replication-controller.md) which will make sure that N pods are running nginx (where in is the number of replicas stated in the spec, which defaults to 1). We also create a [service](services.md) with a selector that matches the replication controller selector. This will configure the kube-proxy load balancers to target pods created by that replication controller on the specified port.
+
+#### docker ps
+
+How do I list what is currently running? Checkout [kubectl get](kubectl/kubectl_get.md).
+
+With docker:
+```
+$ docker ps
+CONTAINER ID        IMAGE               COMMAND                CREATED             STATUS              PORTS                         NAMES
+a9ec34d98787        nginx               "nginx -g 'daemon of   About an hour ago   Up About an hour    0.0.0.0:80->80/tcp, 443/tcp   nginx-app
+```
+
+With kubectl:
+```
+$ kubectl get pods
+NAME              READY     STATUS    RESTARTS   AGE
+nginx-app-5jyvm   1/1       Running   0          1h
+```
+
+#### docker exec
+
+How do I execute a command in a container? Checkout [kubectl exec](kubectl/kubectl_exec.md).
+
+With docker:
+```
+$ docker ps
+CONTAINER ID        IMAGE               COMMAND                CREATED             STATUS              PORTS                         NAMES
+a9ec34d98787        nginx               "nginx -g 'daemon of   8 minutes ago       Up 8 minutes        0.0.0.0:80->80/tcp, 443/tcp   nginx-app
+$ docker exec a9ec34d98787 cat /etc/hostname
+a9ec34d98787
+```
+
+With kubectl:
+```
+$ kubectl get po
+NAME              READY     STATUS    RESTARTS   AGE
+nginx-app-5jyvm   1/1       Running   0          10m
+$ ./kubectl exec nginx-app-5jyvm -- cat /etc/hostname
+nginx-app-5jyvm
+
+```
+
+What about interactive commands?
+
+
+With docker:
+```
+$ docker exec -ti a9ec34d98787 /bin/sh
+# exit
+```
+
+With kubectl:
+```
+$ kubectl exec -ti nginx-app-5jyvm -- /bin/sh      
+#
+```
+
+#### docker logs
+
+How do I follow stdout/stderr of a running process? Checkout [kubectl logs](kubectl/kubectl_logs.md).
+
+
+With docker:
+```
+$ docker logs -f a9e
+192.168.9.1 - - [14/Jul/2015:01:04:02 +0000] "GET / HTTP/1.1" 200 612 "-" "curl/7.35.0" "-"
+192.168.9.1 - - [14/Jul/2015:01:04:03 +0000] "GET / HTTP/1.1" 200 612 "-" "curl/7.35.0" "-"
+```
+
+With kubectl:
+```
+$ kubectl logs -f nginx-app-zibvs
+10.240.63.110 - - [14/Jul/2015:01:09:01 +0000] "GET / HTTP/1.1" 200 612 "-" "curl/7.26.0" "-"
+10.240.63.110 - - [14/Jul/2015:01:09:02 +0000] "GET / HTTP/1.1" 200 612 "-" "curl/7.26.0" "-"
+```
+
+Now's a good time to mention slight difference between pods and containers; by default pods will not terminate if their process's exit. Instead it will restart the process. Then how do I see the logs of the dead process if I can't look at dead pods (because there are none)?
```

Comment:
Docker supports a `--restart=always` option.  

So, another way to explain this, which emphasizes the similarities, is like this:
  - change `docker run nginx` examples above to pass `--restart=always` flag too
  - Here, say "When you run a process as shown in the above examples (via `docker run --restart=always -d` or `kubectl run`), and then processes exits, it will automatically be restarted.    In docker, the output for each invocation of the process concatenated.  But for kubernetes, each invokation is separate.   To see the output from a prevoius run in kubernetes, do this: